radis/api/kuruczapi.py

```diff
@@ -0,0 +1,466 @@
+        """Input Kurucz line list (http://kurucz.harvard.edu/linelists/)
+
+        Args:
+            kuruczf: file path
+
+        Returns:
+            A:  Einstein coefficient in [s-1]
+            nu_lines:  transition waveNUMBER in [cm-1] (#NOT frequency in [s-1])
+            elower: lower excitation potential [cm-1] (#converted from eV)
+            eupper: upper excitation potential [cm-1] (#converted from eV)
+            gupper: upper statistical weight
+            jlower: lower J (rotational quantum number, total angular momentum)
+            jupper: upper J
+            ielem:  atomic number (e.g., Fe=26)
+            iion:  ionized level (e.g., neutral=1, singly)
+            gamRad: log of gamma of radiation damping (s-1) #(https://www.astro.uu.se/valdwiki/Vald3Format)
+            gamSta: log of gamma of Stark damping (s-1)
+            gamvdW:  log of (van der Waals damping constant / neutral hydrogen number) (s-1)
+        """
+        with open(kuruczf) as f:
+            lines = f.readlines()
+        num_lines=len(lines)
+        (
+            wlnmair,
+            loggf,
+            species,
+            elower,
+            jlower,
+            labellower,
+            eupper,
+            jupper,
+            labelupper,
+            gamRad,
+            gamSta,
+            gamvdW,
+            ref,
+            NLTElower,
+            NLTEupper,
+            isonum,
+            hyperfrac,
+            isonumdi,
+            isofrac,
+            hypershiftlower,
+            hypershiftupper,
+            hyperFlower,
+            hypernotelower,
+            hyperFupper,
+            hypternoteupper,
+            strenclass,
+            auto,
+            landeglower,
+            landegupper,
+            isoshiftmA,
+        ) = (
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.array([""] * num_lines, dtype=object),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines),
+            np.zeros(num_lines))
+        ielem, iion = np.zeros(num_lines, dtype=int), np.zeros(num_lines, dtype=int)
+
+        for i, line in enumerate(lines):
```

Comment:
This pure Python loop looks very slow, but Kuruzc database is not huge neither. Does it take time to parse the files?

If so we could use the same structure as when parsing Hitran databases, based on np.fromtxt (as far as i remember) which is C-compiled code

Reply:
Indeed, it does not take much time 
This is how it was originally implemented in Exojax if I can remember well